compiler/rustc_hir_typeck/src/fn_ctxt/checks.rs

```diff
@@ -1656,6 +1656,22 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                     ty::Char => Some(tcx.types.u8),
                     ty::RawPtr(..) => Some(tcx.types.usize),
                     ty::FnDef(..) | ty::FnPtr(..) => Some(tcx.types.usize),
+                    &ty::Pat(base, _) if base.is_integral() => {
+                        let layout = tcx
+                            .layout_of(self.typing_env(self.param_env).as_query_input(ty))
+                            .ok()?;
+                        if layout.uninhabited {
+                            None
+                        } else {
```

Comment:
Patterns types of integers cant be uninhabited can they?

Reply:
even if they were why would we want to fallback to an integer var instead of using the expected (uninhabited) pattern type

Reply:
This PR makes us error when encountering wrapping ranges like 10..2 so that wouldn't be uninhabited as of this PR. Though if it did exist I could imagine it being uninhabited

Reply:
oops yea that was just a leftover from doing various pattern types things in parallel. I turned it into an assert